alf/algorithms/generator.py

```diff
@@ -1067,6 +1177,64 @@ def _rkhs_func_grad(self,
         loss_propagated = torch.sum(grad.detach() * outputs, dim=1)
         return (loss, inverse_mvp_loss), loss_propagated
 
+    def _direct_jac_inverse_vec_prod(self, z, vec):
```

Comment:
I tested direct_jac for GPVI in the generator test ( believe in this case GPVI=GPVI+), and it performs well. It is a just slightly better approximation than GPVI.